components/org.wso2.micro.integrator.coordination/src/main/java/org/wso2/micro/integrator/coordination/RDBMSMemberEventListenerTask.java

```diff
@@ -106,19 +98,12 @@ public RDBMSMemberEventListenerTask(String nodeId, String localGroupId, int hear
             if (wasMemberUnresponsive) {
                 notifyRejoin(nodeID, localGroupId);
                 wasMemberUnresponsive = false;
-                inactiveTimestamp = 0L; // reset in active time stamp
             }
         } catch (Throwable e) {
-            log.warn("Error occurred while reading membership events. ", e);
-            if (inactiveTimestamp.equals(0L)) {
-                inactiveTimestamp = System.currentTimeMillis();
-            }
-            long inactiveHeartbeatAge = System.currentTimeMillis() - inactiveTimestamp;
-            if ((inactiveHeartbeatAge > heartbeatMaxRetryTime) && !wasMemberUnresponsive) {
-                log.warn("Node became unresponsive due to not being able to read events from database");
-                inactiveTimestamp = 0L; // reset in active time stamp
-                wasMemberUnresponsive = true;
+            log.error("Error occurred while reading membership events. ", e);
```

Comment:
Should we do this for any throwable? shouldn't we only do this for and error during, readMembershipEvents()? And print an ERROR/WARN for exceptions during notifyWhatever event?

Reply:
readMembershipEvents throws a run time exception so we can't catch it and do this. Also we have no point in continuing when exception happens in readMembershipEvents.